src/corppa/poetry_detection/annotation/recipe.py

```diff
@@ -0,0 +1,172 @@
+    "custom_theme": {
+        "labels": {
+            # trying to use options from PPA webapp color scheme,
+            # but may not be so great in Prodigy UI.
+            # azure #0788fc seafoam blue #57c4c4 wisteria #9c93c0 pig pink #ed949c
+            "POETRY": "#57c4c4",  # label color for POETRY
+        },
+        "hide_true_newline_tokens": False,
+    },
+    "global_css_dir": CURRENT_DIR,
+}
+
+
+def tokenize_stream(stream, image_prefix=None):
+    """Takes a stream of Prodigy tasks and tokenizes text for span annotation,
+    and optionally adds an image prefix to any image path present.
+    Stream is expected to contain `text` and may contain image_path` and a `meta`
+    dictionary. Returns a generator of the stream.
+    """
+
+    nlp = spacy.blank("en")  # use blank spaCy model for tokenization
+
+    # ensure image prefix does not have a trailing slash
+    if image_prefix is None:
+        image_prefix = ""
+    image_prefix = image_prefix.rstrip("/")
+
+    for task in stream:
+        if task.get("text"):
+            doc = nlp(task["text"])
+            task["tokens"] = [
+                {
+                    "text": token.text,
+                    "start": token.idx,
+                    "end": token.idx + len(token.text),
+                    "id": i,
+                }
+                for i, token in enumerate(doc)
+            ]
+        # add image prefix for serving out images
+        if "image_path" in task:
+            task["image"] = f"{image_prefix}/{task['image_path']}"
+        yield task
+
+
+@recipe(
+    "annotate_text_and_image",
+    dataset=Arg(help="path to input dataset"),
+    labels=Arg("--label", "-l", help="Comma-separated label(s)"),
+    image_prefix=Arg("--image-prefix", "-i", help="Base URL for images"),
+)
+def annotate_text_and_image(
+    dataset: str, source: str, labels: str, image_prefix: str = None
+):
+    """Annotate text and image side by side: allows adding manual spans
+    to both image and text. Intended for page-level annotation.
+    """
+
+    stream = JSONL(source)  # load jsonlines into stream
+    # tokenize for span annotation and add image prefix
+    tokenized_stream = tokenize_stream(stream, image_prefix)
+
+    # split labels by commas and strip any whitespace
+    label_list = [label.strip() for label in labels.split(",")]
+
+    blocks = [
+        {
+            "view_id": "image_manual",
+            "labels": label_list,
+        },
+        {"view_id": "spans_manual", "labels": label_list},
+    ]
+
+    # copy the common config options and add blocks and labels
+    config = PRODIGY_COMMON_CONFIG.copy()
+    config.update(
+        {
+            "blocks": blocks,
+            "labels": label_list,
+            "image_manual_spans_key": "image_spans",
+            # limit image selection to rectangle only, no polygon or freehand
+            "image_manual_modes": ["rect"],
+        }
```

Comment:
We might also want to set the following config options:
- `"darken_image": 0.25` This will darken the area within the bound box by 25% (we might want a different value)
- `"image_manual_show_labels": false` Don't show the labels by default, since this task should only have one by default.

Reply:
...but why do we need a label at all?